cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -225,9 +225,13 @@ func startControllers(c *cloudcontrollerconfig.CompletedConfig, stop <-chan stru
 				}
 			}
 
-			routeController := routecontroller.New(routes, client("route-controller"), c.SharedInformers.Core().V1().Nodes(), c.ComponentConfig.KubeCloudShared.ClusterName, clusterCIDR)
-			go routeController.Run(stop, c.ComponentConfig.KubeCloudShared.RouteReconciliationPeriod.Duration)
-			time.Sleep(wait.Jitter(c.ComponentConfig.GenericComponent.ControllerStartInterval.Duration, ControllerStartJitter))
+			routeController, err := routecontroller.NewRouteController(routes, client("route-controller"), c.SharedInformers.Core().V1().Nodes(), c.ComponentConfig.KubeCloudShared.ClusterName, clusterCIDR)
+			if err != nil {
+				glog.Errorf("Failed to start route controller: %v", err)
```

Comment:
do we really want to survive this error? Why not panic?

Reply:
`panic` looks good. But reference `Service controller` https://github.com/kubernetes/kubernetes/blob/master/cmd/cloud-controller-manager/app/controllermanager.go#L201-L213,  log `error` seems OK